core/src/main/java/io/hyperfoil/core/handlers/json/StreamQueue.java

```diff
@@ -5,118 +5,261 @@
 
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.jctools.util.Pow2;
 
 /**
  * This serves as an abstraction over several {@link ByteStream bytestreams} so that users can just append
  * new buffers to the back and use absolute positioning with single index.
  */
 public class StreamQueue {
-   protected static final Logger log = LogManager.getLogger(StreamQueue.class);
+   private static final Logger log = LogManager.getLogger(StreamQueue.class);
```

Comment:
I've retained this one because if we keep on allocating past a certain threshold we can decide to warn the user of an allocation